code/modules/urist/antagonist/revenant/archetypes/vampiric.dm

```diff
@@ -0,0 +1,449 @@
+	var/suppression_per_bloodsip = BSR_DISTORTION_GROWTH_OVER_SECONDS(30, BSR_DEFAULT_DISTORTION_PER_TICK, BSR_DEFAULT_DECISECONDS_PER_TICK)
+	var/bloodsip_size = 5
+
+	var/removed = FALSE
+	var/added_suppression = 0
+
+	var/datum/reagents/ingested = C.get_ingested_reagents()
+	var/available_amt = ingested.get_reagent_amount(/datum/reagent/blood)
+	var/consume_amt = max(0, available_amt)
+
+	if(consume_amt > 0)
+		removed = ingested.remove_reagent(/datum/reagent/blood, consume_amt)
+		if(removed && bloodsip_size != 0)
+			added_suppression = ((consume_amt / bloodsip_size) * suppression_per_bloodsip)
+			to_chat(src, SPAN_WARNING("You digest all the blood in your stomach... somehow, and use it to anchor yourself to reality a bit better."))
+
+	else
+		to_chat(src, SPAN_WARNING("You currently don't have any blood in your stomach to digest."))
+		return removed
+
+	if(isbsrevenant(src))
+		// We'll allow the proc to run for non-Revenants for funminnery purposes, it
+		// will just not have any positive effects for the user (pure RP)
+		if(src.mind && src.mind.bluespace_revenant)
+			var/datum/bluespace_revenant/revenant_data = src.mind.bluespace_revenant
+
+			if(istype(revenant_data))
+				revenant_data.suppressed_distortion += added_suppression
+
+	return removed
+
+
+/datum/power/revenant/bs_hunger/bloodthirsty
+	flavor_tags = list(
+		BSR_FLAVOR_VAMPIRE,
+		BSR_FLAVOR_DEMONIC,
+		BSR_FLAVOR_OCCULT,
+		BSR_FLAVOR_DEATH,
+		BSR_FLAVOR_GENERIC
+	)
+	activate_message = ("<span class='notice'>You've developed an unnatural thirst for humanoid blood. With an effort of will, you can digest any blood you drink to stabilize your Distortion gain.</span>")
+	name = "Bloodthirsty"
+	isVerb = TRUE
+	verbpath = /mob/proc/revenant_bloodconsume_stomach
+
+
+/mob/proc/revenant_undeadify()
+	set category = "Anomalous Powers"
+	set name = "-Undeadify Yourself-"
+	set desc = "Oops! You shouldn't be seeing this as a verb, please notify admins/coders!"
+
+	src.urist_status_flags |= STATUS_UNDEAD
+	return
+
+
+/datum/power/revenant/bs_power/undead
+	flavor_tags = list(
+		BSR_FLAVOR_VAMPIRE,
+		BSR_FLAVOR_DEMONIC,
+		BSR_FLAVOR_PLAGUE,
+		BSR_FLAVOR_DEATH,
+		BSR_FLAVOR_GENERIC
+	)
+	activate_message = ("<span class='warning'>Something's very wrong... You seem to have lost all vital signs - but somehow it doesn't seem to be much of a problem to you!</span>")
+	name = "Undead"
+	isVerb = FALSE
+	verbpath = /mob/proc/revenant_undeadify
+	distortion_threshold = 30000
+
+
+/datum/power/revenant/distortion/bats
+	flavor_tags = list(
+		BSR_FLAVOR_VAMPIRE,
+		BSR_FLAVOR_DARK
+	)
+	name = "DISTORTION - Bats!"
+
+
+/datum/power/revenant/distortion/bats/Apply(var/atom/A, var/datum/bluespace_revenant/revenant)
+	if(isnull(A) || !istype(A))
+		return
+
+	var/turf/T = get_turf(A)
+	if(!istype(T))
+		return
+
+	var/obj/effect/gateway/hole = new(T)
+	hole.density = 0
```

Comment:
```suggestion
	hole.density = FALSE
```

Reply:
Will change; IIRC this was a copypasta of Baycode for this as of the initial implementation, so I didn't pay much attention.